nestkernel/source.h

```diff
@@ -144,13 +142,13 @@ Source::is_primary() const
 inline void
 Source::disable()
 {
-  gid_ = GID_DISABLED;
+  gid_ = MAX_GID;
 }
 
 inline bool
 Source::is_disabled() const
 {
-  return gid_ == GID_DISABLED;
+  return gid_ == MAX_GID;
```

Comment:
This is counterintuitive. MAX_GID (2^62 -1 ) is not a valid GID, correct? If "high-value" fields are indicate something, for clarity, this should be reflected in the source code.

Reply:
i am not sure i understand. do you mean we should do something `constexpr uint64_t DISABLED = MAX_GID;` and then use that to indicated disabled sources?

Reply:
@jakobj The name of the constant 'MAX_GID' expresses that it represents the highest valid GID, but we test for `gid < MAX_GID`. We hide information here and make the code less readable. Better would be to have two constants.
`GID_DISABLED` and `MAX_GID = GID_DISABLED - 1`. Then we can test for `gid <= MAX_GID` which is what I would expect.